app/src/renderer/vuex/modules/blockchain.js

```diff
@@ -61,6 +81,66 @@ export default ({ commit, node }) => {
         })
       })
     },
+    async queryTxInfo({ state, dispatch }, height) {
+      let blockTxInfo = state.blockTxs.find(
+        b => b.length && b[0].height === height
+      )
+      if (blockTxInfo) {
+        return blockTxInfo
+      }
+      try {
+        blockTxInfo = await dispatch("getTxs", {
+          key: 0,
+          len:
+            state.block && state.block.data && state.block.data.txs
+              ? state.block.data.txs.length
+              : 0,
+          txs:
+            state.block && state.block.data && state.block.data.txs
+              ? state.block.data.txs.slice(0)
+              : []
+        })
+        blockTxInfo && state.blockTxs.push(blockTxInfo)
+        return blockTxInfo
+      } catch (error) {
+        return Promise.reject(error)
+      }
+    },
+    convertTx({ state }, txstring) {
+      let txbytes = b64.toByteArray(txstring)
+      let varintlen = new Uint8Array(varint.encode(txbytes.length))
+      let tmp = new Uint8Array(varintlen.byteLength + txbytes.byteLength)
+      tmp.set(new Uint8Array(varintlen), 0)
+      tmp.set(new Uint8Array(txbytes), varintlen.byteLength)
+      return createHash("ripemd160")
+        .update(Buffer.from(tmp))
+        .digest("hex")
+    },
+    getTxs({ state, commit, dispatch }, { key, len, txs }) {
+      return new Promise(async (resolve, reject) => {
+        if (key >= len) return resolve(txs)
```

Comment:
Could you add a comment stating what this statement accomplishes?

Reply:
👍